CMakeLists.txt

```diff
@@ -34,6 +37,18 @@ if(MSVC)
   set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:MSVCRT")
   set(CMAKE_EXE_LINKER_FLAGS_DEBUG   "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:MSVCRTD")
   set(CMAKE_STATIC_LINKER_FLAGS      "${CMAKE_STATIC_LINKER_FLAGS} /LTCG:incremental")
+else(MSVC)
+
+  # Borrowed from marian.
+  list(APPEND ALL_WARNINGS -Wall; -Werror; -Wextra; -Wno-unused-result; -Wno-deprecated;
+    -Wno-pragmas; -Wno-unused-parameter; -Wno-unused-function;
+    -Wno-unused-value; -Wno-unknown-pragmas; -Wno-sign-compare;
```

Comment:
Unused value is not really useful. It creates more harm than good. At least don't promote it to `Werror`as it is done in the marian codebase. Leave it as a warning and let people fix it as they code. 

Same goes for unused parameter and unused functions and unused result. Let's not go to the marian codebase style of writing a lot of pointless stuff to make a compiler happy (eg https://github.com/marian-nmt/marian-dev/blob/fe74576dc3e957f97b4842951fb6ee7ec52fa324/src/training/communicator.cpp#L53 )

Not sure